packages/tasks/src/model-libraries-snippets.ts

```diff
@@ -403,20 +403,69 @@ backbone = keras_nlp.models.Backbone.from_preset("hf://${model.id}")
 `,
 ];
 
-export const keras_hub = (model: ModelData): string[] => [
-	`# Available backend options are: "jax", "torch", "tensorflow".
+export function keras_hub(model: ModelData): string[] {
+	// If the model has a task.json config, then the base Task class is known
+	let class_name = model.config?.keras_hub_task_json?.class_name;
+	if (!class_name)
+		// If only a config.json is present, the base class will be a "backbone"
+		class_name = model.config?.keras_hub_config_json?.class_name;
+
+	// Fallback heuristic: until task.json is populated in all keras-hub models,
+	// make a best effort, for text-generation models only, to disply
+	// a "XXXCausalLM" base class instead of XXXBackbone.
+	if (model.pipeline_tag == "text-generation" && class_name?.endsWith("Backbone"))
+		class_name = class_name.replace("Backbone", "CausalLM");
```

Comment:
use a succession of ternaries? would be more idiomatic

Reply:
even better: refactored using ?? operator